ccl/backupccl: TestCloudBackupRestoreAzure failed

ccl/backupccl.TestCloudBackupRestoreAzure [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/7670990?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/7670990?buildTab=artifacts#/) on master @ [8548987813ff9e1b8a9878023d3abfc6911c16db](https://github.com/cockroachdb/cockroach/commits/8548987813ff9e1b8a9878023d3abfc6911c16db):


```
=== RUN   TestCloudBackupRestoreAzure
I221124 00:08:04.527901 1 (gostd) rand.go:193  [-] 1  random seed: 1470509626730196725
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/725401e9ce389ccb97d42a5721e8e6a2/logTestCloudBackupRestoreAzure2894590161
    test_log_scope.go:79: use -show-logs to present logs inline
    testcluster.go:384: migration-job-find-already-completed: cannot set fixed timestamp, txn "sql txn" meta={id=78ebd7ab key=/Min pri=0.03151516 epo=0 ts=1669248484.681195056,0 min=1669248484.681195056,0 seq=0} lock=false stat=PENDING rts=1669248484.681195056,0 wto=false gul=1669248485.181195056,0 already performed reads
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/725401e9ce389ccb97d42a5721e8e6a2/logTestCloudBackupRestoreAzure2894590161
--- FAIL: TestCloudBackupRestoreAzure (0.29s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCloudBackupRestoreAzure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-21767

ccl/backupccl.TestCloudBackupRestoreAzure [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/7685233?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/7685233?buildTab=artifacts#/) on master @ [8548987813ff9e1b8a9878023d3abfc6911c16db](https://github.com/cockroachdb/cockroach/commits/8548987813ff9e1b8a9878023d3abfc6911c16db):


```
=== RUN   TestCloudBackupRestoreAzure
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/725401e9ce389ccb97d42a5721e8e6a2/logTestCloudBackupRestoreAzure3839698207
    test_log_scope.go:79: use -show-logs to present logs inline
    testcluster.go:384: migration-job-find-already-completed: cannot set fixed timestamp, txn "sql txn" meta={id=636d1409 key=/Min pri=0.03289301 epo=0 ts=1669334862.384729892,0 min=1669334862.384729892,0 seq=0} lock=false stat=PENDING rts=1669334862.384729892,0 wto=false gul=1669334862.884729892,0 already performed reads
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/725401e9ce389ccb97d42a5721e8e6a2/logTestCloudBackupRestoreAzure3839698207
--- FAIL: TestCloudBackupRestoreAzure (0.39s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCloudBackupRestoreAzure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestCloudBackupRestoreAzure [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/7697458?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/7697458?buildTab=artifacts#/) on master @ [8548987813ff9e1b8a9878023d3abfc6911c16db](https://github.com/cockroachdb/cockroach/commits/8548987813ff9e1b8a9878023d3abfc6911c16db):


```
=== RUN   TestCloudBackupRestoreAzure
I221126 00:06:56.627102 1 (gostd) rand.go:193  [-] 1  random seed: 1095652057335779217
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/725401e9ce389ccb97d42a5721e8e6a2/logTestCloudBackupRestoreAzure2814353713
    test_log_scope.go:79: use -show-logs to present logs inline
    testcluster.go:384: migration-job-find-already-completed: cannot set fixed timestamp, txn "sql txn" meta={id=8ec9e0d4 key=/Min pri=0.02096714 epo=0 ts=1669421216.724946470,0 min=1669421216.724946470,0 seq=0} lock=false stat=PENDING rts=1669421216.724946470,0 wto=false gul=1669421217.224946470,0 already performed reads
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/725401e9ce389ccb97d42a5721e8e6a2/logTestCloudBackupRestoreAzure2814353713
--- FAIL: TestCloudBackupRestoreAzure (0.17s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCloudBackupRestoreAzure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestCloudBackupRestoreAzure [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/7740424?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/7740424?buildTab=artifacts#/) on master @ [8548987813ff9e1b8a9878023d3abfc6911c16db](https://github.com/cockroachdb/cockroach/commits/8548987813ff9e1b8a9878023d3abfc6911c16db):


```
=== RUN   TestCloudBackupRestoreAzure
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/725401e9ce389ccb97d42a5721e8e6a2/logTestCloudBackupRestoreAzure2848103588
    test_log_scope.go:79: use -show-logs to present logs inline
    testcluster.go:384: migration-job-find-already-completed: cannot set fixed timestamp, txn "sql txn" meta={id=85a0f4a4 key=/Min pri=0.01566126 epo=0 ts=1669680422.762279852,0 min=1669680422.762279852,0 seq=0} lock=false stat=PENDING rts=1669680422.762279852,0 wto=false gul=1669680423.262279852,0 already performed reads
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/725401e9ce389ccb97d42a5721e8e6a2/logTestCloudBackupRestoreAzure2848103588
--- FAIL: TestCloudBackupRestoreAzure (0.20s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCloudBackupRestoreAzure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/backupccl/BUILD.bazel
+++ b/pkg/ccl/backupccl/BUILD.bazel
@@ -242,7 +242,6 @@ go_test(
         "//pkg/sql/catalog/descpb",
         "//pkg/sql/catalog/descs",
         "//pkg/sql/catalog/desctestutils",
-        "//pkg/sql/catalog/lease",
         "//pkg/sql/catalog/systemschema",
         "//pkg/sql/catalog/tabledesc",
         "//pkg/sql/execinfra",

--- a/pkg/ccl/backupccl/backup_cloud_test.go
+++ b/pkg/ccl/backupccl/backup_cloud_test.go
@@ -19,7 +19,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/cloud"
 	"github.com/cockroachdb/cockroach/pkg/cloud/amazon"
 	"github.com/cockroachdb/cockroach/pkg/cloud/azure"
-	"github.com/cockroachdb/cockroach/pkg/sql/catalog/lease"
 	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
@@ -51,8 +50,6 @@ func TestCloudBackupRestoreS3(t *testing.T) {
 	defer log.Scope(t).Close(t)
 	creds, bucket := requiredS3CredsAndBucket(t)
 
-	// TODO(dan): Actually invalidate the descriptor cache and delete this line.
-	defer lease.TestingDisableTableLeases()()
 	const numAccounts = 1000
 
 	ctx := context.Background()
@@ -70,8 +67,6 @@ func TestCloudBackupRestoreS3WithLegacyPut(t *testing.T) {
 	defer log.Scope(t).Close(t)
 	creds, bucket := requiredS3CredsAndBucket(t)
 
-	// TODO(dan): Actually invalidate the descriptor cache and delete this line.
-	defer lease.TestingDisableTableLeases()()
 	const numAccounts = 1000
 
 	ctx := context.Background()
@@ -131,8 +126,6 @@ func TestCloudBackupRestoreGoogleCloudStorage(t *testing.T) {
 		skip.IgnoreLint(t, "GOOGLE_BUCKET env var must be set")
 	}
 
-	// TODO(dan): Actually invalidate the descriptor cache and delete this line.
-	defer lease.TestingDisableTableLeases()()
 	const numAccounts = 1000
 
 	ctx := context.Background()
@@ -164,8 +157,6 @@ func TestCloudBackupRestoreAzure(t *testing.T) {
 		skip.IgnoreLint(t, "AZURE_CONTAINER env var must be set")
 	}
 
-	// TODO(dan): Actually invalidate the descriptor cache and delete this line.
-	defer lease.TestingDisableTableLeases()()
 	const numAccounts = 1000
 
 	ctx := context.Background()

